src/main/java/co/inventorsoft/spring_rest_task/service/FileService.java

```diff
@@ -0,0 +1,121 @@
+public class FileService {
+    private final Path fileStorageLocation;
+
+    public FileService(FileStorageProperties fileStorageProperties) {
+        // get path from application.properties
+        this.fileStorageLocation = Paths.get(fileStorageProperties.getUploadDir())
+                .toAbsolutePath().normalize();
+
+        // try to create directories
+        try {
+            Files.createDirectories(this.fileStorageLocation);
+        }
+        catch (Exception ex) {
+            throw new RuntimeException("could not create the directories.", ex);
+        }
+    }
+
+    public String store(MultipartFile file) {
+        // normalize file name
+        String fileName = StringUtils.cleanPath(Objects.requireNonNull(file.getOriginalFilename()));
+
+        try {
+            if (fileName.contains("..")) {
+                throw new RuntimeException("filename " + fileName + " is invalid.");
+            }
+            // copy file to the target directory
+            Path targetDirectory = this.fileStorageLocation.resolve(fileName);
+            Files.copy(file.getInputStream(), targetDirectory, StandardCopyOption.REPLACE_EXISTING);
+
+            return fileName;
+        }
+        catch (IOException e) {
+            throw new RuntimeException("could not store file " + fileName + ".", e);
+        }
+    }
+
+    public Resource load(String fileName) {
+        try {
+            // get path from application.properties and filename
+            Path filePath = fileStorageLocation.resolve(fileName).normalize();
+            Resource resource = new UrlResource(filePath.toUri());
+            if (resource.exists()) {
+                return resource;
+            }
+            else {
+                throw new MyFileNotFoundException("file " + fileName + " not found.");
+            }
+        } catch (MalformedURLException ex) {
+            throw new MyFileNotFoundException("file " + fileName + " not found.", ex);
+        }
+    }
+    public boolean update(String oldFileName, String newFileName) throws IOException {
+        // check if new filename is valid
+        if (newFileName.contains("..")) {
+            return false;
+        }
+        Path filePath = fileStorageLocation.resolve(newFileName).normalize();
+
+        File oldFile = load(oldFileName).getFile();
+        File newFile = new File(filePath.toUri());
+
+        if (!newFile.exists()) {
+            return oldFile.renameTo(newFile);
+        }
+        return false;
+    }
+
+    public boolean delete(String fileName) throws IOException {
+        File file = load(fileName).getFile();
+        return file.delete();
+    }
+
+    /**
+     * Try to determine file's content type
+     */
+    public String getContentType(Resource resource, ServletContext servletContext) {
+        String contentType = null;
+        try {
+            contentType = servletContext.getMimeType(resource.getFile().getAbsolutePath());
+        }
+        catch (IOException ex) {
```

Comment:
Keep **catch** on the same line as the closing bracket
```suggestion
        } catch (IOException ex) {
```